src/olympia/shelves/views.py

```diff
@@ -1,12 +1,25 @@
 from rest_framework import viewsets
+from rest_framework.response import Response
+from rest_framework.views import APIView
 
 from olympia.shelves.models import Shelf
 from olympia.shelves.serializers import ShelfSerializer
 
 
 class ShelfViewSet(viewsets.ModelViewSet):
+    format_kwarg = None
     queryset = Shelf.objects.filter(
         shelfmanagement__enabled=True).order_by('shelfmanagement__position')
-    permission_classes = []
-
     serializer_class = ShelfSerializer
+
+    def get_shelves(self):
+        return self.get_serializer(self.queryset, many=True)
+
+
+class HomepageView(APIView):
+    def get(self, request, format=None):
+        output = {
+            'info': ShelfViewSet(
+                request=request).get_shelves().data,
+        }
+        return Response(output)
```

Comment:
So, we will probably need this kind of combination view in the future when we want to include _different_ types of object in the response, but at the moment all the results will be `Shelf`s, so it doesn't bring us much closer.

Reply:
The addons will be per Shelf, so really the best place for them is inside the Shelf, so that's leads to us including the viewset data inside the Serializer, like you were doing before via requests.